pytorch_lightning/trainer/trainer.py

```diff
@@ -2471,6 +2474,39 @@ def _active_loop(self) -> Optional[Union[FitLoop, EvaluationLoop, PredictionLoop
     Logging properties
     """
 
+    @property
+    def logger(self) -> Optional[LightningLoggerBase]:
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger: Union[LightningLoggerBase, None]) -> None:
+        if new_logger:
+            if isinstance(new_logger, LoggerCollection):
+                self._logger = new_logger
+                self._loggers = [_logger for _logger in new_logger]
```

Comment:
nit: local variables should not be protected

```suggestion
                self._loggers = [logger for logger in new_logger]
```